tools/build-rpms.py

```diff
@@ -0,0 +1,212 @@
+
+
+def msg_error(body):
+    print(f"{fg.ERROR}{fg.BOLD}Error:{fg.RESET} {body}")
+
+
+def msg_info(body):
+    print(f"{fg.INFO}{fg.BOLD}Info:{fg.RESET} {body}")
+
+
+def msg_ok(body):
+    print(f"{fg.OK}{fg.BOLD}OK:{fg.RESET} {body}")
+
+
+def run_command(argv):
+    subprocess.run(argv, check=True)
+
+
+def create_cleanup_function(name, f, *args):
+    def fun():
+        msg_info("Cleaning up: " + name)
+        try:
+            f(*args)
+        except Exception as ex:
+            msg_error("during cleanup: " + str(ex))
+
+    return fun
+
+
+def stage(name, params, fun, *args):
+    msg_info(name + ','.join(params))
+
+    try:
+        ret = fun(*args)
+    except Exception as e:
+        msg_error(f"{name} {','.join(params)} failed: {e}")
+        raise
+
+    msg_ok(f"{name} {','.join(params)}")
+    return ret
+
+
+def create_keypair(cleanup_actions):
+    ec2 = boto3.client('ec2')
+    keyname = f'rpm-builder-{uuid.uuid4()}'
+    response = ec2.create_key_pair(KeyName=keyname)
+    cleanup_actions += [
+        create_cleanup_function(
+            f"keypair {keyname}",
+            lambda k: ec2.delete_key_pair(
+                KeyName=k), keyname)]
+    return keyname, response['KeyMaterial']
+
+
+def create_ec2_instances(cleanup_actions, args, keypair):
+    ec2 = boto3.resource('ec2')
+
+    instances = []
+    for a in args.arch:
+        tags = [
+            {
+                "ResourceType": "instance",
+                "Tags": [
+                    {
+                        "Key": "name",
+                        "Value": f"rpm-builder-{uuid.uuid4()}"
+                    },
+                    {
+                        "Key": "commit",
+                        "Value": f"{args.commit}"
+                    },
+
+                ]
+            }
+        ]
+
+        instance = ec2.create_instances(
+            ImageId=arch_info[a]["ImageId"],
+            MinCount=1,
+            MaxCount=1,
+            InstanceType=arch_info[a]["InstanceType"],
+            KeyName=keypair,
+            TagSpecifications=tags
+        )
+        instances += instance
+
+    for i in instances:
+        cleanup_actions += [
+            create_cleanup_function(
+                f"instance {i.id}",
+                lambda x: x.terminate(),
+                i)]
+        i.wait_until_running()
+        i.reload()
+
+    return instances
+
+
+def setup_ansible(args, instances):
+    with open(os.path.join(args.base_dir, "tools", "appsre-ansible", "inventory"), 'w') as f:
+        f.write("[rpmbuilder]\n")
+        for i in instances:
+            f.write(f"{i.public_ip_address}\n")
+
+
+def run_ansible(args, key_material):
+    os.umask(0)
+    keypath = os.path.join(args.base_dir, "keypair.pem")
+
+    with open(os.open(keypath, os.O_CREAT | os.O_WRONLY, 0o600), 'w') as f:
+        f.write(key_material)
+
+    osbuild_commit = json.load(open(
+        f"{args.base_dir}/Schutzfile"))["rhel-8.6"]["dependencies"]["osbuild"]["commit"]
```

Comment:
isn't `args.base_dir` `pathlib.Path`? If that's the case, then `args.base_dir/"Schutzfile"` should just work and be safer.